aptos-move/framework/aptos-framework/sources/genesis.move

```diff
@@ -102,15 +102,21 @@ module aptos_framework::genesis {
         timestamp::set_time_has_started(&aptos_framework_account);
     }
 
-    /// Genesis step 2: Initialize Aptos coin.
-    fun initialize_aptos_coin(aptos_framework: &signer): MintCapability<AptosCoin> {
+    /// Genesis step 2: Initialize Aptos coin and mint initial balances.
+    fun initialize_aptos_coin(
+        aptos_framework: &signer,
+        initial_balances: vector<InitialBalance>,
+    ): MintCapability<AptosCoin> {
         let (burn_cap, mint_cap) = aptos_coin::initialize(aptos_framework);
         // Give stake module MintCapability<AptosCoin> so it can mint rewards.
         stake::store_aptos_coin_mint_cap(aptos_framework, mint_cap);
 
         // Give transaction_fee module BurnCapability<AptosCoin> so it can burn gas.
         transaction_fee::store_aptos_coin_burn_cap(aptos_framework, burn_cap);
 
+        // Mint initial balances
+        aptos_coin::mint_initial_balances(aptos_framework, initial_balances);
```

Comment:
I don't think we need this function in mint_initial_balances. We can mint directly here with aptos_coin::mint(aptos_framework,...) as the aptos_framework account has mint cap during genesis from line 112 earlier.